### PR TITLE
Release 10.4.0

### PR DESCRIPTION
**Pull request by micbar**
## Description
Merge back Release 10.4.0 branch

## Related Issue
- Closes https://github.com/owncloud/enterprise/issues/3722

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by micbar** [comment]
@phil-davis Let us see what CI says now. I dropped https://github.com/owncloud/core/commit/e506b7ef6571f9350281d91f3f63e5c1fa3f6384

**Comment by phil-davis** [comment]
Looks good - the diff is clean.

**Comment by phil-davis** [comment]
@micbar the "magic commands" that generate `CHANGELOG.md` will need to be run again in this branch, to get the changelogs for those recent dependency updates renamed into the release folder and...

**Review by phil-davis — APPROVED** [review]
Looks all good code that will be fine in master

File: .drone.yml [hunk 1]
```diff
@@ -802,9 +802,9 @@ steps:
   image: owncloudci/php:7.1
   commands:
   - curl -k -s -u user1:123456 -X MKCOL "https://server/remote.php/webdav/new_folder"
-  - curl -k -s -u user1:123456 "https://server/ocs/v2.php/apps/files_sharing/api/v1/shares" --data "path=/new_folder&shareType=0&permissions=15&name=new_folder&shareWith=admin"
+  - "curl -k -s -u user1:123456 \"https://server/ocs/v2.php/apps/files_sharing/api/v1/shares\" -H \"OCS-APIREQUEST: true\" --data \"path=/new_folder&shareType=0&permissions=15&name=new_folder&shareWith=admin\""
   - echo -n "PUBLIC_TOKEN=" > .env
-  - curl -k -s -u user1:123456 "https://server/ocs/v2.php/apps/files_sharing/api/v1/shares" --data "path=/new_folder&shareType=3&permissions=15&name=new_folder" | grep token | cut -d">" -f2 | cut -d"<" -f1 >> .env
+  - "curl -k -s -u user1:123456 \"https://server/ocs/v2.php/apps/files_sharing/api/v1/shares\" -H \"OCS-APIREQUEST: true\" --data \"path=/new_folder&shareType=3&permissions=15&name=new_folder\" | grep token | cut -d\">\" -f2 | cut -d\"<\" -f1 >> .env"
```
Comment (jnweiger) [inline .drone.yml]: The curl commands are now wrapped in doublequotes. And doublequotes within the curl commands need escaping. Why is that?

**Pull request merged**